UserConfigTemplate.txt

```diff
@@ -0,0 +1,773 @@
+      "setvar_DataAcq_RsCorr": {
+        "NumValue": 0
+      },
+      "setvar_DataAcq_RsPred": {
+        "NumValue": 0
+      },
+      "setvar_DataAcq_SSPressure": {
+        "NumValue": 0
+      },
+      "setvar_DataAcq_TerminationDelay": {
+        "NumValue": 0
+      },
+      "setvar_DataAcq_WCC": {
+        "NumValue": 1
+      },
+      "setvar_DataAcq_WCR": {
+        "NumValue": 0
+      },
+      "setvar_DataAcq_dDAQOptOvPost": {
+        "NumValue": 0
+      },
+      "setvar_DataAcq_dDAQOptOvPre": {
+        "NumValue": 0
+      },
+      "setvar_Hardware_Status": {
+        "StrValue": "Independent"
+      },
+      "setvar_Hardware_YokeList": {
+        "StrValue": "Device is not yokeable"
+      },
+      "setvar_Settings_AutoBiasInt": {
+        "NumValue": 1
+      },
+      "setvar_Settings_AutoBiasPerc": {
+        "NumValue": 15
+      },
+      "setvar_Settings_DecMethodFac": {
+        "NumValue": -1
+      },
+      "setvar_Settings_InAirP": {
+        "NumValue": 3.799999952316284
+      },
+      "setvar_Settings_InBathP": {
+        "NumValue": 0.5
+      },
+      "setvar_Settings_InSliceP": {
+        "NumValue": 0.20000000298023224
+      },
+      "setvar_Settings_NearCellP": {
+        "NumValue": 0.6000000238418579
+      },
+      "setvar_Settings_OsciUpdExt": {
+        "NumValue": 10
+      },
+      "setvar_Settings_OsciUpdInt": {
+        "NumValue": 500
+      },
+      "setvar_Settings_Pressure_ADgain": {
+        "NumValue": 0.5
+      },
+      "setvar_Settings_Pressure_DAgain": {
+        "NumValue": 2
+      },
+      "setvar_Settings_SealMaxP": {
+        "NumValue": -1.4
+      },
+      "setvar_Settings_SealStartP": {
+        "NumValue": -0.1
+      },
+      "setvar_Settings_SliceSurfHeight": {
+        "NumValue": 350
+      },
+      "setvar_Settings_SurfaceHeight": {
+        "NumValue": 3500
+      },
+      "setvar_Settings_TPBuffer": {
+        "NumValue": 1
+      },
+      "setvar_Settings_TP_RTolerance": {
+        "NumValue": 1
+      },
+      "slider_DataAcq_ActiveHeadstage": {
+        "NumValue": 1
+      },
+      "tab_DataAcq_Amp": {
+        "NumValue": 1
+      },
+      "tab_DataAcq_Pressure": {
+        "NumValue": 0
+      }
+    },
+    "Test Pulse": {
+      "SetVar_DataAcq_TPAmplitude": {
+        "NumValue": 10
+      },
+      "SetVar_DataAcq_TPAmplitudeIC": {
+        "NumValue": -50
+      },
+      "SetVar_DataAcq_TPBaselinePerc": {
+        "NumValue": 25
+      },
+      "SetVar_DataAcq_TPDuration": {
+        "NumValue": 10
+      }
+    }
+  },
+  "GUI independent configuration data": {
+    "Amplifier Names": "0,1;2,3;4,5;6,7",
+    "Amplifier Serial Numbers": "0;",
+    "Name of Pressure Devices": "Pressure1and2;Pressure3and4;Pressure5and6;Pressure7and8",
+    "Position MCCs": "- none -",
+    "Positive Pressure Calibration Constants": "0.4;0.14;0.05;0.14;0.29;0.03;0.05;0.045",
+    "Save data to": "C:MiesSave",
+    "Stim set file name": ""
+  }
+}
```

Comment:
@t-b can these be moved to the top?

Reply:
When exporting the json information to a string the members are sorted alphabetically. I can give the name of the block "GUI independent configuration data" a prefix that moves it to the top.
The other members at the same level are window names. As long as there are no windows with names that appear alphabetically before the prefix it would stay on top.

Reply:
How about something like "Common configuration data"? That would sort it earlier as "ITC*" and "Dev*" and not look too ulgy.

The technical resason for the sorting is the use of the default std::map in our C++ json library. See https://nlohmann.github.io/json/classnlohmann_1_1basic__json_a5e48a7893520e1314bf0c9723e26ea2a.html#a5e48a7893520e1314bf0c9723e26ea2a for some in-depth technical discussion. It is technical possible to add support for a custom sorting order, but it would be some effort.

Reply:
Ok I first had a name starting with "!", but I set it to your recommended value now.